src/GHC/Util/Unify.hs

```diff
@@ -58,6 +58,12 @@ validSubst' eq = fmap Subst' . mapM f . groupSort . fromSubst'
     where f (x, y : ys) | all (eq y) ys = Just (x, y)
           f _ = Nothing
 
+-- Remove unnecessary brackets from a Subst'. The first argument is a list of unification variables
+-- for which brackets should be removed from their substitutions.
+removeParens :: [LHsExpr GhcPs] -> Subst' (LHsExpr GhcPs) -> Subst' (LHsExpr GhcPs)
```

Comment:
```suggestion
removeParens :: [String] -> Subst' (LHsExpr GhcPs) -> Subst' (LHsExpr GhcPs)
```

Would be better for the map unsafePrettyPrint to call from the caller side, since this API makes no sense if you don't pass a list of strings here.